src/components/Message/renderText/renderText.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { ComponentType } from 'react';
+import ReactMarkdown, { Options, uriTransformer } from 'react-markdown';
+import { find } from 'linkifyjs';
+import uniqBy from 'lodash.uniqby';
+import remarkGfm from 'remark-gfm';
+
+import { Emoji } from './Emoji';
+import { Anchor } from './Anchor';
+import { Mention, MentionProps } from './Mention';
+import { detectHttp, escapeRegExp, matchMarkdownLinks, messageCodeBlocks } from './regex';
+import { emojiMarkdownPlugin, mentionsMarkdownPlugin } from './rehypePlugins';
+
+import type { ReactMarkdownProps } from 'react-markdown/lib/complex-types';
+import type { PluggableList } from 'react-markdown/lib/react-markdown';
+import type { UserResponse } from 'stream-chat';
+import type { DefaultStreamChatGenerics } from '../../../types/types';
+
+export type RenderTextPluginConfigurator = (defaultPlugins: PluggableList) => PluggableList;
+
+export const defaultAllowedTagNames: Array<keyof JSX.IntrinsicElements | 'emoji' | 'mention'> = [
+  'html',
+  'text',
+  'br',
+  'p',
+  'em',
+  'strong',
+  'a',
+  'ol',
+  'ul',
+  'li',
+  'code',
+  'pre',
+  'blockquote',
+  'del',
+  // custom types (tagNames)
+  'emoji',
+  'mention',
+];
+
+function formatUrlForDisplay(url: string) {
+  try {
+    return decodeURIComponent(url).replace(detectHttp, '');
+  } catch (e) {
+    return url;
+  }
+}
+
+function encodeDecode(url: string) {
+  try {
+    return encodeURI(decodeURIComponent(url));
+  } catch (error) {
+    return url;
+  }
+}
+
+const transformLinkUri = (uri: string) => (uri.startsWith('app://') ? uri : uriTransformer(uri));
+
+const getPluginsForward: RenderTextPluginConfigurator = (plugins: PluggableList) => plugins;
+
+export const markDownRenderers: RenderTextOptions['customMarkDownRenderers'] = {
+  a: Anchor,
+  emoji: Emoji,
+  mention: Mention,
+};
+
+export type RenderTextOptions<
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics
+> = {
+  allowedTagNames?: Array<keyof JSX.IntrinsicElements | 'emoji' | 'mention' | string>;
```

Comment:
```suggestion
  allowedTagNames?: Array<keyof JSX.IntrinsicElements | 'emoji' | 'mention' | (string & {})>;
```
This way you get code-completion to work as well. :)

https://stackoverflow.com/a/61048124

Reply:
added and disabled applied eslint rule for now. thank you :)